api/interface.go

```diff
@@ -185,6 +194,11 @@ func NewLoginResultParams(result params.LoginResult) (*LoginResultParams, error)
 type LoginProvider interface {
 	// Login performs log in when connecting to the controller.
 	Login(ctx context.Context, caller base.APICaller) (*LoginResultParams, error)
+	// Token returns the string used for authentication. Some providers may only obtain
```

Comment:
The description should include the errors returned under what circumstances, enclosed in `[ ]` for doc comment references.

And in so doing, we'd for example mention ErrTokenNtAvailable, but the code for some providers (see other comments) would then violate this contract.

Reply:
See other comment - the interface needs to define an expectation that is implemented consistently for every provider. This allows the caller / user of instances which satisfy the interface to know what they are getting and implement the logic accordingly. Otherwise we are breaking all sorts of abstraction / encapsulation concerns.